src/AzureExtension/DataManager/AzureDataManagerCache.cs

```diff
@@ -0,0 +1,296 @@
+// Licensed under the MIT License.
+
+using System.Dynamic;
+using DevHomeAzureExtension.Client;
+using DevHomeAzureExtension.DataManager;
+using DevHomeAzureExtension.DataModel;
+using DevHomeAzureExtension.DeveloperId;
+using Microsoft.TeamFoundation.Core.WebApi;
+using Microsoft.TeamFoundation.SourceControl.WebApi;
+using Microsoft.VisualStudio.Services.Account;
+using Microsoft.VisualStudio.Services.Account.Client;
+using Microsoft.VisualStudio.Services.Common;
+using Microsoft.VisualStudio.Services.WebApi;
+using Serilog;
+
+namespace DevHomeAzureExtension;
+
+public partial class AzureDataManager
+{
+    private static readonly Uri VSAccountUri = new(@"https://app.vssps.visualstudio.com/");
+
+    private static readonly int PullRequestProjectLimit = 50;
+
+    private static readonly int PullRequestRepositoryLimit = 25;
+
+    public async Task UpdateDataForAccountsAsync(RequestOptions? options = null, Guid? requestor = null)
+    {
+        // A parameterless call will always update the data, effectively a 'force update'.
+        await UpdateDataForAccountsAsync(TimeSpan.MinValue, options, requestor);
+    }
+
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+    public async Task UpdateDataForAccountsAsync(TimeSpan olderThan, RequestOptions? options = null, Guid? requestor = null)
+#pragma warning restore CS1998 // Async method lacks 'await' operators and will run synchronously
+    {
+        var cancellationToken = options?.CancellationToken.GetValueOrDefault() ?? default;
+        var requestorGuid = requestor ?? Guid.Empty;
+        var errors = 0;
+        var accountsUpdated = 0;
+        var accountsSkipped = 0;
+        var startTime = DateTime.Now;
+        Exception? firstException = null;
+
+        // Refresh option will clear all sync data, effectively forcing every row to be updated.
+        if (options is not null && options.Refresh)
+        {
+            using var tx = DataStore.Connection!.BeginTransaction();
+            try
+            {
+                _log.Debug("Clearing sync data.");
+                Organization.ClearAllSynced(DataStore);
+            }
+            catch (Exception ex)
+            {
+                _log.Error(ex, "Failed clearing sync data.");
+                tx.Rollback();
+            }
+
+            tx.Commit();
+        }
+
+        var developerIds = DeveloperIdProvider.GetInstance().GetLoggedInDeveloperIdsInternal();
+        foreach (var developerId in developerIds)
+        {
+            _log.Debug($"Updating accounts for {developerId.LoginId}, older than {olderThan}");
+            var accounts = GetAccounts(developerId, cancellationToken);
+            foreach (var account in accounts)
+            {
+                var org = Organization.Get(DataStore, account.AccountUri.ToString());
+                if (org is not null)
+                {
+                    if ((DateTime.Now - org.LastSyncAt) < olderThan)
```

Comment:
Should this use .UTCNow instead?

Reply:
You are correct, there are some interesting and slight nuances when using a timespan and DateTime.Now vs DateTime.UtcNow that can lead to bugs during daylight savings hours and other bizarre moments. I will convert it and the underlying organization timestamps to UtcNow (though they are probably the same since it is stored as ticks, but doesn't hurt to be sure).